rego/rego_test.go

```diff
@@ -1840,6 +1917,189 @@ func TestRegoEvalModulesOnCompiler(t *testing.T) {
 	assertResultSet(t, rs, `[[1]]`)
 }
 
+func TestRegoEvalWithRegoV1(t *testing.T) {
+	tests := []struct {
+		note           string
+		regoVersion    ast.RegoVersion
+		policies       map[string]string
+		query          string
+		expectedResult string
+		expectedErr    string
+	}{
+		{
+			note: "Rego v0",
+			policies: map[string]string{
+				"policy.rego": `package test
+				x[y] { y := 1 }`,
+			},
+			expectedResult: `[[{"x": [1]}]]`,
+		},
+		{
+			note:        "Rego v0, forced v1 compatibility",
+			regoVersion: ast.RegoV0CompatV1,
+			policies: map[string]string{
+				"policy.rego": `package test
+				import rego.v1
+				x contains y if { y := 1 }`,
+			},
+			expectedResult: `[[{"x": [1]}]]`,
+		},
+		{
+			note:        "Rego v0, forced v1 compatibility, invalid rule head",
+			regoVersion: ast.RegoV0CompatV1,
+			policies: map[string]string{
+				"policy.rego": `package test
+				import future.keywords.contains
+				x contains y { y := 1 }`,
+			},
+			expectedErr: "rego_parse_error: `if` keyword is required before rule body",
+		},
+		{
+			note:        "Rego v0, forced v1 compatibility, missing required imports",
+			regoVersion: ast.RegoV0CompatV1,
+			policies: map[string]string{
+				"policy.rego": `package test
+				x contains y if { y := 1 }`,
+			},
+			expectedErr: "rego_parse_error: var cannot be used for rule name", // FIXME: Improve error message
```

Comment:
Is this something to address in this PR?

Reply:
I believe this might require a larger set of changes in the parser. Let's not delay this PR and risk it not getting in this release, and fix this in the next release instead.